### DISCUSSION TITLE
Add support for custom seperator in variant group.

### DISCUSSION THREAD

**Discussion by 1821746019** [ben-rogerson/twin.macro#871] [Ideas]
### The prettier-tailwindcss-plugin will cannot work well with variant group syntax. It tamper the original logic.

```jsx
<DrumMachineBox css={[tw`landscape:(w-full aspect-video max-w-4xl )`]} />

```
wiil be formatted to
```jsx
<DrumMachineBox css={[tw`landscape:(w-full ) aspect-video max-w-4xl`]} />
```
### So, I think this feature request is reasonable.
![image](https://github.com/user-attachments/assets/91243a15-a253-4899-b753-45e2df7ec3ae)

https://x.com/adamwathan/status/1849509712368226792

```jsx
<DrumMachineBox css={[tw`landscape:(w-full,aspect-video,max-w-4xl)`]} />
```
![image](https://github.com/user-attachments/assets/dbd67436-a65d-47eb-ba35-64bd27d47c9d)

**Comment by ben-rogerson** [comment]
Hey there - thanks for letting me know about that poll.
Great to see variant grouping might be coming to Tailwind!

I'm hesitant to implement the feature ahead of a Tailwind release - they may still change the syntax + theres a chance it'll "just work" once it's released in Tailwind as [Twin uses some of the Tailwind codebase](https://github.com/ben-rogerson/twin.macro/blob/master/src/core/lib/util/twImports.ts).

**Comment by 1821746019** [comment]
Hi Ben,

Thanks for your response. I completely understand your hesitation about implementing this feature before Tailwind’s official release. I agree that the syntax might change, and Twin will likely sync up with Tailwind’s final version once it's out.

That said, my main concern is that `prettier-plugin-tailwindcss` currently formats the code incorrectly with the variant group syntax. It would be really helpful if Twin could support a custom separator, specifically using a comma (,) instead of a space, as the separator for variant groups. This would prevent Prettier from misformatting the code and make it easier for users adopting the new syntax.

If this could be done in a non-breaking way, it would significantly improve the user experience.

Looking forward to your thoughts!
